static/docs/commands-reference/pull.md

```diff
@@ -84,11 +84,11 @@ reflinks or hardlinks to put it in the workspace without copying. See
   behavior from the `dvc checkout` command because `dvc pull` in effect performs
   a _checkout_ after downloading files.
 
-- `-R`, `--recursive` - `targets` values is expected to be a directory path.
-  Determines the files to download by searching the named directory and its
-  subdirectories for DVC files to download data for. Along with providing a
-  `target`, or `target` along with `--with-deps` it is yet another way to cut
-  the scope of DVC files to download.
+- `-R`, `--recursive` - each `targets` value is expected to be a path to a file 
+  or directory, relative to current location(eg.`data/models`). Determines 
+  DVC-files to download data for the target directory. Along with providing a 
+  `target`, and/or `--with-deps` it is yet another way to cut the scope of 
+  DVC-files to download.
```

Comment:
@kurianbenoy it's better. can you explain me what does this option mean in your own words, please?  :) It's still hard to read, I still don't understand from this description what's going on.

Reply:
For `dvc pull -R`
1. Targets can be a file or directory
2. What it does is pulls cache for new DVC-files from the subdirectories inside DVC remote storage 
3. --with-deps cuts scopes of files being pulled

Reply:
All true, but emphasizing it properly is important.

1. It was made specifically for a case when target is a directory (to search inside them for the DVC-files). It accepts files just for simplicity. So, that you don't have to run a separate command if you have a mix of both. The way we describe it should match to the primary case - when people are looking for a recursive search inside a dir.
2. It pulls data from the cache to the workspace (and downloads them from remote first if needed, in a way it is described in this document) for data files and directories that are referenced in those DVC-files. I'm not sure I understand the _new_ part in your sentence.
3.  `Along with providing a  target, and/or --with-deps it is yet another way to cut the scope of  DVC-files to download.` - let's just remove it completely, I don't see any value in it to be honest.